File: lista_4/huaauhahhuahau.py
risada = str(input())
vogais = ['a','e','i','o','u']
re = []
rd = []

for i in risada:
      if i in vogais:
            re.append(i)

rd = re[:]
rd.reverse()

tamanho = len(re)
confere = 0

for j in range(tamanho):
      if rd[j] == re[j]:
            confere += 1
      else:
            confere += 0

if confere == tamanho:
      print('S')
else:
      print('N')

File: lista_4/huaauhahhuahau.rb
risada = gets.chomp.to_s.chars
vogais = ['a','e','i','o','u']

for r in risada
	if r not in vogais
		risada.delete(r)
	end
end

re = risada
rd = risada.reverse

if (re == rd)
      print('S')
else:
      print('N')
